src/spaces/productspace.jl

```diff
@@ -255,3 +245,16 @@ Base.eltype(P::ProductSpace) = eltype(typeof(P))
 
 Base.IteratorEltype(::Type{<:ProductSpace}) = Base.HasEltype()
 Base.IteratorSize(::Type{<:ProductSpace}) = Base.HasLength()
+
+# Promotion and conversion
+# ------------------------
+function Base.promote_rule(::Type{S}, ::Type{<:ProductSpace{S}}) where {S<:ElementarySpace}
+    return ProductSpace{S}
+end
+
+# ProductSpace to ElementarySpace
+Base.convert(::Type{S}, P::ProductSpace{S,0}) where {S<:ElementarySpace} = oneunit(S)
+Base.convert(::Type{S}, P::ProductSpace{S}) where {S<:ElementarySpace} = fuse(P.spaces...)
```

Comment:
Is that something you want to have happen automatically?

Reply:
Although, I guess since the promote is giving you `ProductSpace` this would never be called "implicitly".

Reply:
I think this is mostly a convenience thing, in principle `ProductSpace` is the "larger" type, so this should indeed never be called